src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/XamlIlBindingPathHelper.cs

```diff
@@ -37,6 +37,10 @@ public static IXamlType UpdateCompiledBindingExtension(AstTransformationContext
                 bindingResultType = transformed.BindingResultType;
                 binding.Arguments[0] = transformed;
             }
+            if (binding.Arguments.Count > 0 && binding.Arguments[0] is XamlIlBindingPathNode alreadyTransformed)
```

Comment:
Should probably be `else if`

Reply:
Shouldn't be a problem here, but yes, better to fix